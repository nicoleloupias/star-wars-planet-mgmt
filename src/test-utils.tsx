import { ChakraProvider } from "@chakra-ui/react";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import React from "react";
import { Router, RouterProvider } from "react-router-dom";
import { theme } from "./theme";
const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <ChakraProvider theme={theme}>
      <Router location={history.location} navigator={history}>
        {children}
      </Router>
    </ChakraProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {}
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => <AllProviders history={history}>{children}</AllProviders>,
    ...options
  });

export { customRender as render };
