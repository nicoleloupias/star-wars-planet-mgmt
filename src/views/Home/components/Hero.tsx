import { Box, Flex, Heading, Img, Text } from "@chakra-ui/react";
import darth from "../../../assets/darth.png";
import { ArrowDownIcon } from "@chakra-ui/icons";

export const Hero = () => {
  return (
    <Flex
      as="section"
      overflow="hidden"
      w="full"
      minH="100vh"
      bg="red.500"
      alignItems="center"
      py={20}
      pos="relative"
      _after={{
        content: `""`,
        width: "full",
        height: "full",
        bgImage: darth,
        pos: "absolute",
        top: 0,
        left: 0,
        bgSize: "100%",
        bgPosition: "center",
        bgRepeat: "no-repeat",
        mixBlendMode: "multiply"
      }}
    >
      <Box zIndex={1} ml={{ base: 4, lg: 12 }} alignSelf="stretch" w="1px" bgColor="white"></Box>
      <Heading
        zIndex={1}
        fontWeight={900}
        justifySelf="center"
        fontSize={{ base: "4rem", lg: "6rem" }}
        textTransform="uppercase"
        pos="absolute"
        left="50%"
        width="max-content"
        transform="translateX(-50%)"
      >
        Explore the galaxy
      </Heading>
      <Box mr={{ base: 4, lg: 12 }} ml="auto" alignSelf="stretch" w="1px" bgColor="white"></Box>

      <Text fontSize="xl" zIndex={1} pos="absolute" bottom={20} left={"50%"} transform={"translateX(-50%)"}>
        Scroll
        <ArrowDownIcon ml={2} />
      </Text>
    </Flex>
  );
};
