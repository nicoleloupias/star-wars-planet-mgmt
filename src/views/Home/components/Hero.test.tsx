import { describe, expect, test } from "vitest";
import { render } from "../../../test-utils";
import { Hero } from "./Hero";

describe("<Hero/>", () => {
  test("should render succesfully", () => {
    const component = render(<Hero />);

    expect(component).toMatchSnapshot();
  });

  test("should render elements correctly", () => {
    const component = render(<Hero />);

    expect(component.getByText(/Explore the galaxy/)).toBeInTheDocument();
    expect(component.getByText(/Scroll/)).toBeInTheDocument();
    expect(component.getByRole("img")).toHaveAttribute("src", "/src/assets/Darth-Vader-PNG-HD-Image.png");
  });
});
