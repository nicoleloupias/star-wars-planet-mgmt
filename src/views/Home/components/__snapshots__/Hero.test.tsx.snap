// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Hero/> > should render succesfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-dark"
  >
    <div>
      <section
        class="css-1lbat96"
      >
        <div
          class="css-ybip2u"
        />
        <h2
          class="chakra-heading css-s2chw8"
        >
          Explore the galaxy
        </h2>
        <div
          class="css-1etnvhi"
        />
        <p
          class="chakra-text css-1is01ss"
        >
          Scroll
        </p>
        <img
          class="chakra-image css-odqkwc"
          src="/src/assets/Darth-Vader-PNG-HD-Image.png"
        />
      </section>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <section
      class="css-1lbat96"
    >
      <div
        class="css-ybip2u"
      />
      <h2
        class="chakra-heading css-s2chw8"
      >
        Explore the galaxy
      </h2>
      <div
        class="css-1etnvhi"
      />
      <p
        class="chakra-text css-1is01ss"
      >
        Scroll
      </p>
      <img
        class="chakra-image css-odqkwc"
        src="/src/assets/Darth-Vader-PNG-HD-Image.png"
      />
    </section>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
